optimum/habana/transformers/trainer.py

```diff
@@ -1510,6 +1515,12 @@ def evaluation_loop(
                 )
                 self.already_wrapped_for_hpu_graphs = True
 
+        # if full fp16 or bf16 eval is wanted and this ``evaluation`` or ``predict`` isn't called
+        # while ``train`` is running, cast it to the right dtype first and then put on device
+        if not self.is_in_train:
+            if args.bf16_full_eval:
+                model = model.to(dtype=torch.bfloat16, device=args.device)
+
```

Comment:
maybe we can add code for args.fp16_full_eval also (for future if we decide to support fp16 also).